app/services/ui_notifications/clean_expired.rb

```diff
@@ -0,0 +1,40 @@
+module UINotifications
+  class CleanExpired
+    def initialize(options = {})
+      if options[:blueprint].present?
+        blueprint = NotificationBlueprint.find_by(name: options[:blueprint])
+        raise(Foreman::Exception, 'blueprint name does not exist') if blueprint.nil?
+        @blueprint = blueprint.name
+      end
+      parsed_time = options[:expired_at].is_a?(String) ? options[:expired_at].to_time : Time.now.utc
+      raise(Foreman::Exception, 'Expired time cannot be greater then now') if parsed_time > Time.now.utc
+      @expired_at = parsed_time
+      @group = options[:group]
+    end
+
+    def clean!
+      UINotifications::Base.logger.info("Removing all expired notifications")
+      to_delete = Notification.joins(:notification_blueprint)
+                      .where('notifications.expired_at < ?', @expired_at)
+                      .where(blueprint_query)
+                      .where(group_query)
+      @deleted_count = to_delete.destroy_all.length
```

Comment:
@tbrisker I didn't use `delete_all` because according to [documentation](https://apidock.com/rails/ActiveRecord/Base/delete_all/class), dependent rules defined on associations are not honored, and error may raised.

So to use `delete_all` on notifications I've delete associated `notification_recipients` first.